api/user/schema.py

```diff
@@ -112,11 +113,46 @@ def mutate(self, info, email, **kwargs):
         exact_user = query_user.filter(UserModel.email == email).first()
         if not exact_user:
             raise GraphQLError("User not found")
-        user_role = UsersRole.query.filter_by(user_id=exact_user.id).first()
-        new_role = kwargs.pop('role_id')
-        user_role.role_id = new_role
-        with SaveContextManager(user_role, 'Role id', new_role):
-            return ChangeUserRole(user=exact_user)
+
+        if len(exact_user.roles) < 1:
+            raise GraphQLError('user has no role')
+
+        new_role_obj = RoleModel.query.filter_by(id=kwargs['role_id']).first()
+        if not new_role_obj:
+            raise GraphQLError('invalid role id')
+
+        exact_user.roles[0] = new_role_obj
+        exact_user.save()
+        return ChangeUserRole(user=exact_user)
+
+
+class CreateUserRole(graphene.Mutation):
+
+    class Arguments:
+        user_id = graphene.Int(required=True)
+        role_id = graphene.Int(required=True)
+    user_role = graphene.Field(User)
+
+    def mutate(self, info, **kwargs):
+        user = User.get_query(info)
+        exact_user = user.filter_by(id=kwargs['user_id']).first()
+
+        if not exact_user:
+            raise GraphQLError('user not found')
+
+        role = Role.get_query(info)
+        exact_role = role.filter_by(id=kwargs['role_id']).first()
+
+        if not exact_role:
+            raise GraphQLError('Role id does not exist')
+
+        if len(exact_user.roles) > 0:
```

Comment:
I wouldn't recommend this method of check. Use a simpler method by simply checking if the list in non-empty without necessarily using the len method and the greater than symbol.